src/output.c

```diff
@@ -768,3 +778,55 @@ handle_output_power_manager_set_mode(struct wl_listener *listener, void *data)
 		break;
 	}
 }
+
+void
+virtual_output_add(struct server *server, const char *output_name)
+{
+	if (output_name) {
+		/*
+		 * Prevent creating outputs with the same name
+		 */
+		struct output *output;
+		wl_list_for_each(output, &server->outputs, link) {
+			if (wlr_output_is_headless(output->wlr_output)) {
+				if (!strcmp(output->wlr_output->name, output_name)) {
+					wlr_log(WLR_DEBUG,
+					"refusing to create virtual output with duplicate name");
+					return;
+				}
+			}
+		}
+		strncpy(server->headless.pending_output_name, output_name,
+				sizeof(server->headless.pending_output_name));
+	} else {
+		server->headless.pending_output_name[0] = '\0';
+	}
+	wlr_headless_add_output(server->headless.backend, 1920, 1080);
```

Comment:
Should we hard-code 1920x1080 here?

Reply:
I thought about it. In fact, i tried to hardcode `0, 0` so the user would then decide. But wlroots is quirky - setting it to `0, 0` here disallows changing resolution from tools like wlr-randr (returns error).

Any ideas what would be a better default value?

Reply:
Ah makes sense then. 1920x1080 is as good as any.

Be good to add a comment to that effect for future devs.
`/* Setting it to 0, 0 here disallows changing resolution from tools like wlr-randr (returns error) */`